adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -162,8 +169,12 @@ private void handleEndpointConnection(final MqttEndpoint endpoint) {
             endpoint.reject(MqttConnectReturnCode.CONNECTION_REFUSED_SERVER_UNAVAILABLE);
 
         } else {
-
-            endpoint.publishHandler(message -> {
+            final AtomicBoolean clientConnected = new AtomicBoolean(false);
```

Comment:
The MqttEndpoint has the client state (connected or not) but it's not exposed through the interface. I can expose it now making available to you. if you want you can merge this change and then I'll update with the new property from the MqttEndpoint.